src/main/scala/mesosphere/marathon/state/PathId.scala

```diff
@@ -167,7 +231,7 @@ object PathId {
     */
   private def childOf(parent: PathId): Validator[PathId] = {
     isTrue[PathId](s"Identifier is not child of $parent. Hint: use relative paths.") { child =>
-      !parent.absolute || (child.canonicalPath(parent).parent == parent)
+      !parent.absolute || (child.canonicalPath(parent.asAbsolutePath).parent == parent)
```

Comment:
Funny. I had issue with `canonicalPath` before. What if `child` is absolute? I'm not sure what canonical means then. Eg what if the parent is `/dev/ci` and child is `/prod/db`. What is the result of `child.canonicalPath(parent.asAbsolutePath)`?